apps/remix-ide/src/app/tabs/compile-tab.js

```diff
@@ -90,6 +90,13 @@ class CompileTab extends ViewPlugin {
    */
 
   listenToEvents () {
+    this.on('filePanel', 'setWorkspace', (workspace) => {
+      this.compileTabLogic.isHardhatProject().then((result) => {
+        if (result) this.compilerContainer.hardhatCompilation.style.display = 'flex'
```

Comment:
instead of hiding, maybe this should be disabled.

Reply:
There may be more checkboxes in future and only disabling them can take a lot of place. So as discussed, I will leave it like that